inject/src/main/java/io/micronaut/inject/annotation/DefaultAnnotationMetadata.java

```diff
@@ -1934,18 +1999,27 @@ public static void contributeDefaults(AnnotationMetadata target, AnnotationMetad
             source = source.getDeclaredMetadata();
         }
         if (target instanceof DefaultAnnotationMetadata && source instanceof DefaultAnnotationMetadata) {
-            final Map<String, Map<CharSequence, Object>> existingDefaults = ((DefaultAnnotationMetadata) target).annotationDefaultValues;
+            DefaultAnnotationMetadata damTarget = (DefaultAnnotationMetadata) target;
+            DefaultAnnotationMetadata damSource = (DefaultAnnotationMetadata) source;
+            final Map<String, Map<CharSequence, Object>> existingDefaults = damTarget.annotationDefaultValues;
             if (existingDefaults != null) {
-                final Map<String, Map<CharSequence, Object>> additionalDefaults = ((DefaultAnnotationMetadata) source).annotationDefaultValues;
+                final Map<String, Map<CharSequence, Object>> additionalDefaults = damSource.annotationDefaultValues;
                 if (additionalDefaults != null) {
                     existingDefaults.putAll(
                             additionalDefaults
                     );
                 }
             } else {
-                final Map<String, Map<CharSequence, Object>> additionalDefaults = ((DefaultAnnotationMetadata) source).annotationDefaultValues;
+                final Map<String, Map<CharSequence, Object>> additionalDefaults = damSource.annotationDefaultValues;
                 if (additionalDefaults != null) {
-                    additionalDefaults.forEach(((DefaultAnnotationMetadata) target)::addDefaultAnnotationValues);
+                    additionalDefaults.forEach(damTarget::addDefaultAnnotationValues);
+                }
+            }
+            if (damSource.repeated != null && !damSource.repeated.isEmpty()) {
+                if (damTarget.repeated == null) {
+                    damTarget.repeated = new HashMap<>(damSource.repeated);
```

Comment:
should be a linked hashmap to retain order?

Reply:
No, it's converted to sorted string array